### PR TITLE
Forward EPICS alarm status and severity

### PR DESCRIPTION
**Pull request by mchorazak**
### Issue

Part of DM-872
Closes DM-1667

### Description of work

- Updated `conanfile.txt` to use a new version of the schema package that contains `f142` schema with alarm status and severity.
- Every PV is now forwarded with alarm status.
- If the alarm status didn't change since the previous update then the flatbuffer message contains the default value of `NO_CHANGE`.
- Periodic updates from the cache always contain the alarm status because the file writer will use these as the first value in `NXlog`s
- Updated `EpicsClientRandom.cpp` to populate the `alarm` field to pass system tests.

- _Entire f142.cpp needs unit tests and the architecture needs significant work to make it testable, therefore will be a separate ticket for it._

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

**Comment by matthew-d-jones** [comment]
Forwarder only needs to include alarm status in the flatbuffer message if it has changed, this will be addressed in DM-1667.

File: src/schemas/f142/f142.cpp [hunk 1]
```diff
@@ -347,6 +347,30 @@ Value_t makeValue(flatbuffers::FlatBufferBuilder &Builder,
   return {Value::NONE, 0};
 }
 
+std::unique_ptr<AlarmStatus>
+getAlarmInfo(epics::pvData::PVStructurePtr const &PVStructureField) {
+  auto AlarmField = PVStructureField->getSubField("alarm");
+  auto MessageField = ((epics::pvData::PVStructure *)(AlarmField.get()))
```
Comment (mattclarke) [inline src/schemas/f142/f142.cpp]: can we avoid the C-style cast?

File: src/schemas/f142/f142.cpp [hunk 2]
```diff
@@ -347,6 +347,30 @@ Value_t makeValue(flatbuffers::FlatBufferBuilder &Builder,
...
+                          ->getSubField("message");
+  auto MessageString = static_cast<epics::pvData::PVScalarValue<std::string> *>(
+                           MessageField.get())
+                           ->get();
+  // Message field is HIHI_ALARM, LOW_ALARM, etc. We have to drop _ALARM in
+  // every case apart from NO_ALARM
+  if (MessageString.compare("NO_ALARM") != 0)
+    MessageString = MessageString.substr(0, MessageString.length() - 6);
+
+  auto StatusNames = EnumNamesAlarmStatus();
```
Comment (mattclarke) [inline src/schemas/f142/f142.cpp]: I think this needs some comments to explain what is going on

**Comment by matthew-d-jones** [comment]
Waiting on https://github.com/ess-dmsc/streaming-data-types/pull/46

**Pull request merged**